downstairs/src/region.rs

```diff
@@ -605,11 +649,46 @@ impl Region {
             }
         };
         save_stream_to_file(extent_copy, repair_stream.into_inner()).await?;
+        let mut count = 1;
+
+        // If we are replacing our region with one from an older version
+        // that contained the SQLite, then look for and copy the database
+        // files.
+        for opt_file in &[ExtentType::Db, ExtentType::DbShm, ExtentType::DbWal]
+        {
+            let filename = extent_file_name(eid as u32, opt_file.clone());
+
+            if repair_files.contains(&filename) {
+                let extent_shm = Self::create_copy_file(
+                    copy_dir.clone(),
+                    eid,
+                    Some(opt_file.clone()),
+                )?;
+                let repair_stream = match repair_server
+                    .get_extent_file(eid as u32, opt_file.to_file_type())
+                    .await
+                {
+                    Ok(rs) => rs,
+                    Err(e) => {
+                        crucible_bail!(
```

Comment:
There's a race condition here if the repair server cleans up the shm / wal files before you get to them, but that's probably fine...

Reply:
👍🏻 not a concern, because "the other Downstairs decides to be adversarial" isn't really in our system model